test/runtests.jl

```diff
@@ -33,4 +54,7 @@ if HDF5.has_parallel()
     include("mpio.jl")
 end
 
+# Clean up after all resources
+HDF5.API.h5_close()

```

Comment:
Need to clean up after external.jl

Reply:
Can we remove this change and because the external file issue is a separate bug.  I propose  pushing this is a separate commit.    (Logically trying to keep  PRs to a related set of features/bugfixes)

Reply:
I put it into #882